src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -795,66 +1065,66 @@ function AnalyticsService:LogProgression(data: ProgressionEvent): { [any]: any }
 		elseif data.score ~= nil and typeof(data.score) ~= "number" then
 			return reject("score must be a number")
 		end
-		
-		GameAnalytics:addProgressionEvent(data.userId, {
+
+		self:_wrapper("addProgressionEvent", data.userId, {
 			progressionStatus = GameAnalytics.EGAProgressionStatus[data.status],
 			progression01 = data.progression01,
 			progression02 = data.progression02,
 			progression03 = data.progression03,
-			score = data.score or 0
+			score = data.score or 0,
 		})
-		
-		return resolve()
+
+		resolve()
```

Comment:
```suggestion
		return resolve()
```
The linter/type checker complains if there's a return type defined for a function, but the promise is resolved/rejected without returning the function (For some reason)? It's an invalid error since we know we're returning a promise, but the code checker flags it for some reason.